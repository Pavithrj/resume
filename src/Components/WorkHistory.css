.workhistory {
    margin-left: 10px;
}

.workhistorydetails {
    margin-left: 10px;
    display: flex;
    justify-content: space-evenly;
}

.workhistorytimeperiod {
    width: 100px;
}

.workdetails {
    display: flex;
    flex-direction: column;
    margin-left: 10px;
    width: 400px;
}

.position {
    width: 220px;
    font-weight: bold;
}

.place {
    font-weight: lighter;
    width: 275px;
}