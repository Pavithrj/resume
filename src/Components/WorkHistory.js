import { useContext } from 'react';
import './WorkHistory.css';
import ProfileContext from './ProfileContext';

function WorkHistory() {
    const profileData = useContext(ProfileContext);
    const { workHistory } = profileData;
    return (
        <div className='workhistorycontaner'>
            <h3 className="workhistory">Work Experience</h3>
            <div>
                {workHistory.map((work) => {
                    return (
                        <div className="workhistorydetails">
                            <div className='workhistorytimeperiod'>
                                {work.timePeriod}
                            </div>
                            <div className='workdetails'>
                                <div className='position'>
                                    {work.position}
                                    <br />
                                    <div className='place'>
                                        {work.place}
                                    </div>
                                    <br />
                                </div>
                                {work.description}
                            </div>
                        </div>
                    )
                })}
            </div>
        </div>
    )
}

export default WorkHistory;