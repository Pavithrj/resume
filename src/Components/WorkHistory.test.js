import { render } from "@testing-library/react";
import ProfileContext from './ProfileContext';
import profileData from '../Details.json';
import WorkHistory from "./WorkHistory";

describe('Contact Component', () => {
    const customRender = (ui, { providerProps, ...renderOptions }) => {
        return render(
            <ProfileContext.Provider value={profileData}>{ui}</ProfileContext.Provider>,
            renderOptions,
        )
    }
    it('should print contact details', () => {
        const { asFragment } = customRender(<WorkHistory />, {
            value: profileData
        });
        expect(asFragment()).toMatchSnapshot();
    })
})