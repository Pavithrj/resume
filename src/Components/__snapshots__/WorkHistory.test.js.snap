// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should print contact details 1`] = `
<DocumentFragment>
  <div
    class="workhistorycontaner"
  >
    <h3
      class="workhistory"
    >
      Work History
    </h3>
    <div>
      <div
        class="workhistorydetails"
      >
        <div
          class="workhistorytimeperiod"
        >
          Sep 2020 - Current
        </div>
        <div
          class="workdetails"
        >
          <div
            class="position"
          >
            Principal Software Engineer
            <br />
            <div
              class="place"
            >
              Mindbody, Pune
            </div>
            <br />
          </div>
          Implemented a design system with a component library using React JS, Typescript, and Material UI to make the UI consistent across the product.
        </div>
      </div>
      <div
        class="workhistorydetails"
      >
        <div
          class="workhistorytimeperiod"
        >
          Dec 2018 - Sept 2020
        </div>
        <div
          class="workdetails"
        >
          <div
            class="position"
          >
            Sr. Software Engineer
            <br />
            <div
              class="place"
            >
              Infogain
            </div>
            <br />
          </div>
          Migrated legacy Backbone JS UI to more maintainable React JS for e-commerce website LampsPlus.
        </div>
      </div>
      <div
        class="workhistorydetails"
      >
        <div
          class="workhistorytimeperiod"
        >
          Nov 2017 - Dec 2018
        </div>
        <div
          class="workdetails"
        >
          <div
            class="position"
          >
            Sr. Solution Developer
            <br />
            <div
              class="place"
            >
              SpadeWorx Software Services
            </div>
            <br />
          </div>
          Built a SharePoint app using Vanilla JS with Angular JS to create intranet portals within a few minutes.
        </div>
      </div>
      <div
        class="workhistorydetails"
      >
        <div
          class="workhistorytimeperiod"
        >
          Dec 2013 - Nov 2017
        </div>
        <div
          class="workdetails"
        >
          <div
            class="position"
          >
            Associate
            <br />
            <div
              class="place"
            >
              Cognizant Technology Solutions, PUNE
            </div>
            <br />
          </div>
          Built intranet portal for Novartis using React JS and SharePoint for 100K+ employees across the globe.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
